pluginframework/converters/converters_test.go

```diff
@@ -0,0 +1,228 @@
+func (f *TestEnum) Set(v string) error {
+	switch v {
+	case `TEST_ENUM_A`, `TEST_ENUM_B`, `TEST_ENUM_C`:
+		*f = TestEnum(v)
+		return nil
+	default:
+		return fmt.Errorf(`value "%s" is not one of "TEST_ENUM_A", "TEST_ENUM_B", "TEST_ENUM_C"`, v)
+	}
+}
+
+func (f *TestEnum) Type() string {
+	return "MonitorInfoStatus"
+}
+
+type DummyNestedTfSdk struct {
+	Name    types.String `tfsdk:"name" tf:"optional"`
+	Enabled types.Bool   `tfsdk:"enabled" tf:"optional"`
+}
+
+type DummyGoSdk struct {
+	Enabled           bool                        `json:"enabled"`
+	Workers           int64                       `json:"workers"`
+	Floats            float64                     `json:"floats"`
+	Description       string                      `json:"description"`
+	Tasks             string                      `json:"tasks"`
+	Nested            *DummyNestedGoSdk           `json:"nested"`
+	NoPointerNested   DummyNestedGoSdk            `json:"no_pointer_nested"`
+	NestedList        []DummyNestedGoSdk          `json:"nested_list"`
+	NestedPointerList []*DummyNestedGoSdk         `json:"nested_pointer_list"`
+	Map               map[string]string           `json:"map"`
+	NestedMap         map[string]DummyNestedGoSdk `json:"nested_map"`
+	Repeated          []int64                     `json:"repeated"`
+	Attributes        map[string]string           `json:"attributes"`
+	EnumField         TestEnum                    `json:"enum_field"`
+	AdditionalField   string                      `json:"additional_field"`
+	DistinctField     string                      `json:"distinct_field"` // distinct field that the tfsdk struct doesn't have
+	ForceSendFields   []string                    `json:"-"`
+}
+
+type DummyNestedGoSdk struct {
+	Name            string   `json:"name"`
+	Enabled         bool     `json:"enabled"`
+	ForceSendFields []string `json:"-"`
+}
+
+// Function to construct individual test case with a pair of matching tfSdkStruct and gosdkStruct.
+// Verifies that the conversion both ways are working as expected.
+func RunConverterTest(t *testing.T, description string, tfSdkStruct DummyTfSdk, goSdkStruct DummyGoSdk) {
+	convertedGoSdkStruct := DummyGoSdk{}
+	assert.True(t, !TfSdkToGoSdkStruct(context.Background(), tfSdkStruct, &convertedGoSdkStruct).HasError())
+	assert.True(t, reflect.DeepEqual(convertedGoSdkStruct, goSdkStruct), fmt.Sprintf("tfsdk to gosdk conversion - %s", description))
+
+	convertedTfSdkStruct := DummyTfSdk{}
+	assert.True(t, !GoSdkToTfSdkStruct(context.Background(), goSdkStruct, &convertedTfSdkStruct).HasError())
+	assert.True(t, reflect.DeepEqual(convertedTfSdkStruct, tfSdkStruct), fmt.Sprintf("gosdk to tfsdk conversion - %s", description))
+}
+
+var tests = []struct {
+	name        string
+	tfSdkStruct DummyTfSdk
+	goSdkStruct DummyGoSdk
+}{
+	{
+		"string conversion",
+		DummyTfSdk{Description: types.StringValue("abc")},
+		DummyGoSdk{Description: "abc", ForceSendFields: []string{"Description"}},
+	},
+	{
```

Comment:
Can we add one test case for DummyTfSdk value set to zero values for each of the primitive types and verifying that ForceSendFields is set properly?

Reply:
Sounds good, added